### ISSUE TITLE
Something is wrong with the field statistics of the term vector request

### ISSUE THREAD

**Issue by ngocdb** [Smile-SA/elasticsuite#520]
Hello,
I am using Smile ES with Magento 2.1.8 version.
When the index has no documents, I got the error as below.

Not sure what is wrong with "The spellcheck request" ? (term_statistics in getTermVectors)

Please help me to understand this issue.

```
2 exception(s):
Exception #0 (Elasticsearch\Common\Exceptions\ServerErrorResponseException): illegal_state_exception: Something is wrong with the field statistics of the term vector request: Values are 
org.elasticsearch.common.xcontent.XContentBuilderString@2463c36d 0
org.elasticsearch.common.xcontent.XContentBuilderString@7463797c 0
org.elasticsearch.common.xcontent.XContentBuilderString@26c0c1de 0
Exception #1 (Elasticsearch\Common\Exceptions\ServerErrorResponseException): {"error":{"root_cause":[{"type":"illegal_state_exception","reason":"Something is wrong with the field statistics of the term vector request: Values are \norg.elasticsearch.common.xcontent.XContentBuilderString@2463c36d 0\norg.elasticsearch.common.xcontent.XContentBuilderString@7463797c 0\norg.elasticsearch.common.xcontent.XContentBuilderString@26c0c1de 0"}],"type":"illegal_state_exception","reason":"Something is wrong with the field statistics of the term vector request: Values are \norg.elasticsearch.common.xcontent.XContentBuilderString@2463c36d 0\norg.elasticsearch.common.xcontent.XContentBuilderString@7463797c 0\norg.elasticsearch.common.xcontent.XContentBuilderString@26c0c1de 0"},"status":500}

Exception #0 (Elasticsearch\Common\Exceptions\ServerErrorResponseException): illegal_state_exception: Something is wrong with the field statistics of the term vector request: Values are 
org.elasticsearch.common.xcontent.XContentBuilderString@2463c36d 0
org.elasticsearch.common.xcontent.XContentBuilderString@7463797c 0
org.elasticsearch.common.xcontent.XContentBuilderString@26c0c1de 0
#0 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(644): Elasticsearch\Connections\Connection->tryDeserializeError(Array, 'Elasticsearch\\C...')
#1 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(607): Elasticsearch\Connections\Connection->tryDeserialize500Error(Array)
#2 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(265): Elasticsearch\Connections\Connection->process5xxError(Array, Array, Array)
#3 /media/datas/websites/newoutlets/vendor/react/promise/src/FulfilledPromise.php(25): Elasticsearch\Connections\Connection->Elasticsearch\Connections\{closure}(Array)
#4 /media/datas/websites/newoutlets/vendor/guzzlehttp/ringphp/src/Future/CompletedFutureValue.php(55): React\Promise\FulfilledPromise->then(Object(Closure), NULL, NULL)
#5 /media/datas/websites/newoutlets/vendor/guzzlehttp/ringphp/src/Core.php(341): GuzzleHttp\Ring\Future\CompletedFutureValue->then(Object(Closure), NULL, NULL)
#6 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(283): GuzzleHttp\Ring\Core::proxy(Object(GuzzleHttp\Ring\Future\CompletedFutureArray), Object(Closure))
#7 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(159): Elasticsearch\Connections\Connection->Elasticsearch\Connections\{closure}(Array, Object(Elasticsearch\Connections\Connection), Object(Elasticsearch\Transport), Array)
#8 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php(99): Elasticsearch\Connections\Connection->performRequest('POST', '/magento2_defau...', Array, '{"doc":{"spelli...', Array, Object(Elasticsearch\Transport))
#9 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Endpoints/AbstractEndpoint.php(79): Elasticsearch\Transport->performRequest('POST', '/magento2_defau...', Array, Array, Array)
#10 /media/datas/websites/newoutlets/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Client.php(531): Elasticsearch\Endpoints\AbstractEndpoint->performRequest()
#11 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Spellchecker.php(149): Elasticsearch\Client->termvectors(Array)
#12 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Spellchecker.php(83): Smile\ElasticsuiteCore\Search\Adapter\Elasticsuite\Spellchecker->getTermVectors(Object(Smile\ElasticsuiteCore\Search\Spellchecker\Request))
#13 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Adapter/Elasticsuite/Spellchecker.php(66): Smile\ElasticsuiteCore\Search\Adapter\Elasticsuite\Spellchecker->loadSpellingType(Object(Smile\ElasticsuiteCore\Search\Spellchecker\Request))
#14 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Request/Builder.php(192): Smile\ElasticsuiteCore\Search\Adapter\Elasticsuite\Spellchecker->getSpellingType(Object(Smile\ElasticsuiteCore\Search\Spellchecker\Request))
#15 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-core/Search/Request/Builder.php(145): Smile\ElasticsuiteCore\Search\Request\Builder->getSpellingType(Object(Smile\ElasticsuiteCore\Search\Request\ContainerConfiguration), 'hello')
#16 /media/datas/websites/newoutlets/app/code/Miragedesign/Elasticproduct/Model/ResourceModel/Product/Fulltext/Collection.php(686): Smile\ElasticsuiteCore\Search\Request\Builder->create(1, 'quick_search_co...', 0, 0, 'hello', Array, Array, Array, Array)
#17 /media/datas/websites/newoutlets/app/code/Miragedesign/Elasticproduct/Model/ResourceModel/Product/Fulltext/Collection.php(332): Miragedesign\Elasticproduct\Model\ResourceModel\Product\Fulltext\Collection->loadProductCounts()
#18 /media/datas/websites/newoutlets/var/generation/Miragedesign/Elasticproduct/Model/ResourceModel/Product/Fulltext/Collection/Interceptor.php(167): Miragedesign\Elasticproduct\Model\ResourceModel\Product\Fulltext\Collection->getProductCountByAttributeSetId()
#19 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-catalog/Block/Navigation.php(155): Miragedesign\Elasticproduct\Model\ResourceModel\Product\Fulltext\Collection\Interceptor->getProductCountByAttributeSetId()
#20 /media/datas/websites/newoutlets/vendor/smile/elasticsuite/src/module-elasticsuite-catalog/Block/Navigation.php(142): Smile\ElasticsuiteCatalog\Block\Navigation->addFacets()
#21 /media/datas/websites/newoutlets/vendor/magento/framework/View/Element/AbstractBlock.php(264): Smile\ElasticsuiteCatalog\Block\Navigation->_prepareLayout()
#22 /media/datas/websites/newoutlets/vendor/magento/framework/View/Layout/Generator/Block.php(139): Magento\Framework\View\Element\AbstractBlock->setLayout(Object(Magento\Framework\View\Layout\Interceptor))
...
```

**Comment by ngocdb** [comment]
If there is at least one document in the index, the error gone.

**Comment by genyuk** [comment]
Hi. 
Have the similar issue since we moved to Magento 2.1.8 CE and updated elasticsearch using composer.

Quick search is not working, throwing this error message:
/en/search/ajax/suggest/?q=m.&_=1504199919603

There has been an error processing your request

illegal_state_exception: Something is wrong with the field statistics of the term vector request: Values are 
org.elasticsearch.common.xcontent.XContentBuilderString@9fe26d4 0
org.elasticsearch.common.xcontent.XContentBuilderString@7d42dbff 0
org.elasticsearch.common.xcontent.XContentBuilderString@13d6f9a6 0

Elasticsearch indexes reindex without errors.

Elastic Search v 2.1.3


Please advise.
Thank you

**Comment by romainruaud** [comment]
Hello @ngocdb and @genyuk 

I am not able to reproduce your issue on a fresh installed Magento 2.1.8 with latest Elasticsuite.

Which version of Elasticsearch are your using ?

**Comment by genyuk** [comment]
Hello 
We have :
Magento 2.1.8
ElasticSearch: 2.4.6.
"lucene_version" : "5.5.4"
Smile/Elasticsuite: 2.4.x-dec

Tried to reinstall ElasticSearch with complete wipe out of ElasticSearch and indexes with no luck.
It was working ok before the update of Magento & Elasticsuite.

I can provide an access to our development server in private message.

Here is what we are seeing in the elasticsearch log:
What could cause this issue?

[2017-09-01 10:12:04,241][WARN ][rest.suppressed          ] path: /svm_en_catalog_category/category//_termvectors, params: {dfs=true, term_statistics=true, index=svm_en_catalog_category, type=category}
java.lang.IllegalStateException: Something is wrong with the field statistics of the term vector request: Values are 
org.elasticsearch.common.xcontent.XContentBuilderString@9fe26d4 0
org.elasticsearch.common.xcontent.XContentBuilderString@7d42dbff 0
org.elasticsearch.common.xcontent.XContentBuilderString@13d6f9a6 0
	at org.elasticsearch.action.termvectors.TermVectorsResponse.buildFieldStatistics(TermVectorsResponse.java:322)
	at org.elasticsearch.action.termvectors.TermVectorsResponse.buildField(TermVectorsResponse.java:205)
	at org.elasticsearch.action.termvectors.TermVectorsResponse.toXContent(TermVectorsResponse.java:194)
	at org.elasticsearch.rest.action.support.RestToXContentListener.buildResponse(RestToXContentListener.java:46)
	at org.elasticsearch.rest.action.support.RestToXContentListener.buildResponse(RestToXContentListener.java:41)
	at org.elasticsearch.rest.action.support.RestToXContentListener.buildResponse(RestToXContentListener.java:33)
	at org.elasticsearch.rest.action.support.RestResponseListener.processResponse(RestResponseListener.java:43)
	at org.elasticsearch.rest.action.support.RestActionListener.onResponse(RestActionListener.java:49)
	at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:89)
	at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:85)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:235)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:221)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:838)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:822)
	at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:812)
	at org.elasticsearch.transport.DelegatingTransportChannel.sendResponse(DelegatingTransportChannel.java:58)
	at org.elasticsearch.transport.RequestHandlerRegistry$TransportChannelWrapper.sendResponse(RequestHandlerRegistry.java:140)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$ShardTransportHandler.messageReceived(TransportSingleShardAction.java:283)
	at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$ShardTransportHandler.messageReceived(TransportSingleShardAction.java:275)
	at org.elasticsearch.transport.TransportRequestHandler.messageReceived(TransportRequestHandler.java:33)
	at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:77)
	at org.elasticsearch.transport.TransportService$4.doRun(TransportService.java:378)
	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by genyuk** [comment]
Any tips how to fix this?

The primary search is still working, "virtual category" addon is also working.
Elastic server is in green state. 
There are hundreds of documents in the index.
Here is some more details for Elastic Server running:

 elasticsearch.service - Elasticsearch
   Loaded: loaded (/usr/lib/systemd/system/elasticsearch.service; enabled)
 Main PID: 15892 (java)
   CGroup: /system.slice/elasticsearch.service

Sep 08 17:39:24 vps78163.vps.ovh.ca systemd[1]: Started Elasticsearch.
Sep 08 17:39:26 vps78163.vps.ovh.ca elasticsearch[15892]: [2017-09-08 17:39:26,470][INFO ][node                     ] [Americop] version[2.4.6], pid[15892], build[5376dca/2017-07-18T12:17:44Z]
Sep 08 17:39:26 vps78163.vps.ovh.ca elasticsearch[15892]: [2017-09-08 17:39:26,476][INFO ][node                     ] [Americop] initializing ...
Sep 08 17:39:27 vps78163.vps.ovh.ca elasticsearch[15892]: [2017-09-08 17:39:27,442][INFO ][plugins                  ] [Americop] modules [reindex, lang-expression, lang-groovy], plugins [analysis-phonetic, analysis-icu], sites []
Sep 08 17:39:27 vps78163.vps.ovh.ca elasticsearch[15892]: [2017-09-08 17:39:27,565][INFO ][env                      ] [Americop] using [1] data paths, mounts [[/ (rootfs)]], net usable_space [5.5gb], net total_space [19... types [rootfs]
Sep 08 17:39:27 vps78163.vps.ovh.ca elasticsearch[15892]: [2017-09-08 17:39:27,568][INFO ][env                      ] [Americop] heap size [1007.3mb], compressed ordinary object pointers [true]

Server health:
epoch      timestamp cluster    status node.total node.data shards pri relo init unassign pending_tasks max_task_wait_time active_shards_percent
1504906791 17:39:51  svmelastic green           1         1     18  18    0    0        0             0                  -                100.0%

Nodes:
host      ip        heap.percent ram.percent load node.role master name
127.0.0.1 127.0.0.1           10          89 0.68 d         *      Americop


Indices:
health status index                                         pri rep docs.count docs.deleted store.size pri.store.size
green  open   svm_xamsa_en_thesaurus_20170908_210753          1   0          0            0       159b           159b
green  open   svm_fr_thesaurus_20170908_210752                1   0          0            0       159b           159b
green  open   svm_xamsa_fr_catalog_product_20170908_210750    1   0          8            0     20.2kb         20.2kb
green  open   svm_xamsa_fr_catalog_category_20170908_210752   1   0        286            0    158.4kb        158.4kb
green  open   svm_halo_fr_thesaurus_20170908_210753           1   0          0            0       159b           159b
green  open   svm_fr_catalog_product_20170908_210746          1   0       8353           12      4.5mb          4.5mb
green  open   svm_xamsa_en_catalog_product_20170908_210750    1   0          8            0     20.2kb         20.2kb
green  open   svm_halo_fr_catalog_product_20170908_210750     1   0         12            0     34.4kb         34.4kb
green  open   svm_halo_en_catalog_product_20170908_210745     1   0         12            0     34.4kb         34.4kb
green  open   svm_en_catalog_category_20170908_210751         1   0        286            0    158.7kb        158.7kb
green  open   svm_xamsa_en_catalog_category_20170908_210752   1   0        286            0    158.4kb        158.4kb
green  open   svm_en_catalog_product_20170908_210740          1   0       8353           12      4.6mb          4.6mb
green  open   svm_halo_fr_catalog_category_20170908_210751    1   0        286            0    158.4kb        158.4kb
green  open   svm_halo_en_thesaurus_20170908_210752           1   0          0            0       159b           159b
green  open   svm_en_thesaurus_20170908_210752                1   0          0            0       159b           159b
green  open   svm_xamsa_fr_thesaurus_20170908_210753          1   0          0            0       159b           159b
green  open   svm_halo_en_catalog_category_20170908_210751    1   0        286            0    158.4kb        158.4kb
green  open   svm_fr_catalog_category_20170908_210751         1   0        286            0    175.6kb        175.6kb




Any tip will be appreciated. Thank you

**Comment by ngocdb** [comment]
We have :
Magento 2.1.8
ElasticSearch: 2.4.5.
"lucene_version" : "5.5.4"
Smile/Elasticsuite: 2.4.x-dec

As said, if there is at least one document in the index, the error gone.

**Comment by genyuk** [comment]
Our category index does have some records as we have a lot of categories. For now we have disabled search in autocomplete  for categories, but we loose ability to search in category index.

Is it possible to fix that?

**Comment by stefanguapa** [comment]
@romainruaud they mean, if you try to search on a storeview which doesn't have a product, it'll return this error. This **usually** won't occur, but may someone have a webshop which is in progress and contains storeviews without products, the search will fail.

**Comment by afoucret** [comment]
PR #566 fix this one and is available on branches master, 2.3.x and 2.4.x
